app/javascript/lib/rich_text/unfurl/lib/opengraph_embed_operation.js

```diff
@@ -14,10 +14,35 @@ export default class OpengraphEmbedOperation {
   }
 
   perform() {
+    // Check if unfurling is enabled via environment variable
+    if (!this.#isUnfurlingEnabled()) {
+      console.log("URL unfurling is disabled via ENABLE_URL_UNFURLING environment variable")
+      return Promise.resolve()
+    }
+
+    console.log(`Attempting to unfurl URL: ${this.url}`)
+    
     return this.#createOpenGraphMetadataRequest()
-      .then(response => response.json)
-      .then(this.#insertOpengraphAttachment.bind(this))
-      .catch(() => null)
+      .then(response => {
+        console.log(`Unfurl response for ${this.url}:`, response.status, response.statusText)
+        
+        if (!response.ok) {
+          console.warn(`Failed to fetch OpenGraph metadata for ${this.url}: ${response.status} ${response.statusText}`)
+          return null
+        }
+        return response.json
+      })
+      .then(data => {
+        if (data) {
+          console.log(`Successfully unfurled ${this.url}:`, data)
+          this.#insertOpengraphAttachment(data)
+        } else {
+          console.log(`No data returned for ${this.url}`)
+        }
+      })
+      .catch(error => {
+        console.warn(`Error unfurling URL ${this.url}:`, error)
+      })
```

Comment:
This looks like the same code just with logging?
Maybe it would be simpler to just log the error in the catch branch?